rmw_zenoh_cpp/src/detail/buffer_pool.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "buffer_pool.hpp"
+
+#include <cstddef>
+#include <cstdlib>
+#include <mutex>
+#include <vector>
+
+#include "rcutils/allocator.h"
+#include "rcutils/env.h"
+#include "logging_macros.hpp"
+
+namespace rmw_zenoh_cpp
+{
+///=============================================================================
+BufferPool::BufferPool()
+: buffers_(), mutex_()
+{
+  const char * env_value;
+  const char * error_str = rcutils_get_env("RMW_ZENOH_BUFFER_POOL_MAX_SIZE_BYTES", &env_value);
```

Comment:
Can you please put a note somewhere in https://github.com/ros2/rmw_zenoh/blob/rolling/docs/design.md explaining what this environment variable is and what it does?  I know that document is out-of-date w.r.t. the changes to zenoh-cpp, but I still think it has value to have this documented somewhere.